docs/spec/spec.md

```diff
@@ -0,0 +1,950 @@
+
+**RPC Using Direct Return**
+
+Ballerina CLI generates the relevant service skeleton, and the implementation of the unary RPC call using direct return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function GetFeature(Point point) returns Feature|error {
+        foreach Feature feature in FEATURES {
+            if feature.location == point {
+                return feature;
+            }
+        }
+        return {location: point, name: ""};
+    }
+}
+```
+
+Here, the RPC implementation creates a featured record and directly return it from the remote method.
+
+**RPC Using a Caller**
+
+The Ballerina implementation of the same unary RPC using a caller is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function GetFeature(RouteGuideFeatureCaller caller, Point point) returns error? {
+        Feature?|error feature = featureFromPoint(point);
+        if feature is Feature {
+            check caller->sendFeature(feature);
+        } else if feature is error {
+            check caller->sendError(<grpc:Error> feature);
+        } else {
+            check caller->sendFeature({location: {latitude: 0, longitude: 0}, name: ""});
+        }
+    }
+}
+```
+
+## 4.2. Server Streaming RPC
+The RPC service definition of a server streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc ListFeatures(Rectangle) returns (stream Feature) {}
+}
+```
+
+**RPC Using Direct Return**
+
+The Ballerina implementation of the server streaming RPC using a direct return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function ListFeatures(Rectangle rectangle) returns stream<Feature, grpc:Error?>|error {
+
+        Feature[] selectedFeatures = [];
+        foreach Feature feature in FEATURES {
+            if inRange(feature.location, rectangle) {
+                selectedFeatures.push(feature);
+            }
+        }
+        return selectedFeatures.toStream();
+    }
+}
+```
+
+**RPC Using a Caller**
+
+The Ballerina implementation of the server streaming RPC using a caller return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function ListFeatures(RouteGuideFeatureCaller caller, Rectangle rectangle) returns error? {
+
+        foreach Feature feature in FEATURES {
+            if inRange(feature.location, rectangle) {
+                check caller->sendFeature(feature);
+            }
+        }
+    }
+}
+```
+
+## 4.3. Client Streaming RPC
+The RPC service definition of a client streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc RecordRoute(stream Point) returns (RouteSummary) {}
```

Comment:
```suggestion
    rpc RecordRoute(stream Point) returns (RouteSummary) {};
```

Reply:
The semicolon is not needed https://github.com/grpc/grpc-go/blob/v1.38.0/examples/route_guide/routeguide/route_guide.proto